x-pack/plugin/fleet/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/fleet/40_secrets_get.yml

```diff
@@ -3,6 +3,10 @@
       fleet.post_secret:
         body: '{"value": "test secret"}'
   - set:  { id: id }
+  # search node needs to be available for fleet.get_secret to work
+  - do:
+      cluster.health:
+        wait_for_active_shards: all
```

Comment:
nit I preferred green. But if this works, also fine.

Reply:
why out of curiosity? I'd thought maybe `wait_for_active_shards` was a bit more precise about the condition it required.

Reply:
So I was not sure if "active" would mean it considers it active if only the primary is fine but not the replica (since in stateful if the primary is fine, both ingestion and searches can be processed). But looking now at the code of how it calculates the wait_for_active_shards, I see

```
if (response.getUnassignedShards() == 0 && response.getInitializingShards() == 0) {
                    // if we are waiting for all shards to be active, then the num of unassigned and num of initializing shards must be 0
                    waitForCounter++;
                }
```

So it means that will check that the search shard is neither unassigned nor initializing, so it covers exactly what we want! I think it's fine to use wait_for_active_shards.

Reply:
ok, I'll flip it back :)